matrix-sdk/src/main/java/org/matrix/androidsdk/fragments/MatrixMessageListFragment.java

```diff
@@ -955,6 +979,18 @@ public void onSuccess(Void info) {
                         @Override
                         public void run() {
                             onMessageSendingSucceeded(event);
+
+                            if (mFutureReadMarkerEvent != null && prevEventId.equals(mFutureReadMarkerEvent)) {
```

Comment:
did you test the case in which the event is echoed before the request gets the response ?
i don't see anything on this implementation.